### PR TITLE
dune-sync-5 Update Readme after Parse Config

### LINKED ISSUE #5: Update Readme after Parse Config
The Local Development instructions in the README will change once we parse config files (i.e. query id wont live in the env). Make sure to update the readme when ready.

Specifically, this is blocked by #4 - but shouldn't be forgotten.

_Originally posted by @bh2smith in https://github.com/bh2smith/dune-sync/pull/3#discussion_r1798336419_

### PR DESCRIPTION
**Pull request by mooster531**
Closes #5

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
Beauty. Just wondering if we are sure this is going to be the final config... would be annoying if it changed and we had to remember to come back here. But otherwise I think it's ready to go.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,78 @@
...
+- `poll_frequency`: Integer. Determines how often, in seconds, to poll Dune for results after running the given query ID
+- `parameters`: Optional Dune Query parameters.
+    - `name`: String. Name of parameter
+    - `type`: String. Type of parameter. Must be one of: "text", "number", "datetime", "enum"
+    - `value`: Any. Value to send for the defined parameter
+
+#### Postgres
+
+- `ref`: String. Must be exactly `postgres`
+- `query_string`: String. Query to run against a Postgres server in order to fetch results and submit them to a
```
Comment (bh2smith) [inline README.md:37]: We might want to make it possible to run a query file. I suspect some queries might be very long strings. In this case, the user would have to mount the config file along with, say a directory "queries" containing a bunch of raw SQL.

https://github.com/bh2smith/dune-sync/issues/44

File: README.md [hunk 2]
```diff
@@ -1,11 +1,78 @@
...
+#### Sources
+
+Currently supported sources are Dune and PostgreSQL. Description of required parameters follows.
+
+#### Dune
+
+- `ref`: String. Must be exactly `dune`
+- `query_id`: Integer. ID of an existing Dune Query to execute and fetch results from.
+- `query_engine`: String and one of exactly `medium` or `large`.
+- `poll_frequency`: Integer. Determines how often, in seconds, to poll Dune for results after running the given query ID
```
Comment (bh2smith) [inline README.md:28]: These should be somehow marked as optional with defaults "medium" and "1"

File: README.md [hunk 3]
```diff
@@ -1,11 +1,78 @@
...
+
+```shell
+docker build -t dune-sync .
+docker run --rm -v "$(pwd)/config.yaml:/app/config.yaml" --env-file .env dune-sync
+```
 
 ## Local Development
 
-Fill out the empty fields in [Sample Env](.env.sample) (`DUNE_API_KEY` and `QUERY_ID`)
+Fill out the empty fields in [Sample Env](.env.sample) (`DUNE_API_KEY` and `DB_URL`)
```
Comment (bh2smith) [inline README.md:75]: I wonder if the end user might have multiple postgres instances. Maybe we could do something like 

`DB_URL_*`

and then the config could reference which to use by number?

https://github.com/bh2smith/dune-sync/issues/45

**Comment by bh2smith** [comment]
Going to merge as is and tag a release candidate. Opened a few issues that might be worth including.

File: README.md (hunk 1 above)
Comment (mooster531) [reply README.md:37]: Query file could be supported but we might also suggest using Yaml Heredoc syntax for that, like this:
```yaml
  - name: Write Results of Job 1 back
    source:
      ref: postgres
      if_exists: "append"
      query_string: |
        this
        will
        be
        some
        very
        long
        query

    destination:
      ref: dune
      table_name: "dune_sync_test_table"
```

File: README.md (hunk 3 above)
Comment (mooster531) [reply README.md:75]: I'd rather just have the DSN as a parameter in the `source` / `destination` blocks, like:
```yaml
    destination:
      ref: postgres
      table_name: "test_table"
      if_exists: "replace"
      dsn: "postgresql://postgres:postgres@localhost:5672/dbname"

```

**Pull request merged**